lib/lcevc/lcevc_dil.js

```diff
@@ -0,0 +1,281 @@
+      typeof libDPIModule !== 'undefined' &&
+      LcevcDil.SupportStatus;
+  }
+
+  /**
+   * Create canvas for LCEVC Dil to render if not exists.
+   * @private
+   */
+  createCanvas_() {
+    this.canvas_ = /** @type {HTMLCanvasElement} */
+      (shaka.util.Dom.createHTMLElement('canvas'));
+    if (LcevcDil.SupportObject.webGLSupport(this.canvas_)) {
+      this.lcevcContainer_ = shaka.util.Dom.createHTMLElement('div');
+      this.lcevcContainer_.style.position = 'relative';
+
+      if (this.dilConfig_['playerControls']) {
+        this.videoControls_ = shaka.util.Dom.getElementById(
+            this.dilConfig_['playerControls']['controlsID']);
+        this.videoControls_.id =
+        this.dilConfig_['playerControls']['controlsID'];
+        this.lcevcContainer_.appendChild(this.videoControls_);
+      }
+
+      this.canvas_.style.position = 'absolute';
+      this.media_.style.display = 'none';
+
+      this.parent_ = this.media_.parentElement;
+      if (this.parent_) {
+        this.lcevcContainer_.appendChild(this.canvas_);
+        this.parent_.appendChild(this.lcevcContainer_);
+        this.resizeObserver_ = new ResizeObserver((_entries) => {
+          window.requestAnimationFrame(() => {
+            this.playerResize_(this.lcevcContainer_, this.dil_,
+                this.media_, this.canvas_);
+          });
+        });
+
+        this.resizeObserver_.observe(this.parent_);
+      }
+    }
+  }
+
+  /**
+   * Resizing the container based on the source video dimensions.
+   * @private
+   */
+  playerResize_(lcevcContainer, lcevcDil, mediaElement, canvas) {
+    const lcevcContainerParent = /** @type {HTMLElement} */
+      (lcevcContainer.parentElement);
+
+    if (this.isSupported()) {
+      // We try to find the actual aspect ratio with the lcevc enhanced frame
+      // if this value is not a valid value we default it to 1920x1080
+      const aspectRatio = (lcevcDil && lcevcDil.aspectRatio > 0 ?
+        lcevcDil.aspectRatio :
+        mediaElement.videoWidth / mediaElement.videoHeight) || 1920 / 1080;
+      lcevcContainer.style.width =
+        `${(lcevcContainerParent).offsetWidth}px`;
+      if (lcevcDil && lcevcDil.isFullscreen) {
+        lcevcContainer.style.height =
+          `${window.innerHeight - this.videoControls_.offsetHeight}px`;
+
+        if (canvas.height < lcevcContainer.offsetHeight) {
+          this.canvas_.style.top =
+          `${(window.innerHeight - canvas.height) / 2}px`;
+        } else {
+          canvas.style.top = '';
+          canvas.style.height =
+            `${window.innerHeight - this.videoControls_.offsetHeight}px`;
+        }
+      } else {
+        lcevcContainer.style.height =
+          `${lcevcContainerParent.offsetWidth / aspectRatio}px`;
+
+        canvas.style.height = `${lcevcContainer.offsetHeight}px`;
+        canvas.style.top = '';
+      }
+    }
+  }
+
+  /**
+   * Set styles for video controls
+   * @private
+   * */
+  setControlsStyle_() {
```

Comment:
Are these the video controls from our UI?  If so, you shouldn't be styling them here.  Can you explain this a bit?

Reply:
This was part of the library controls which in the case of shaka player is not needed. Hence removed.